module/module.go

```diff
@@ -108,37 +115,53 @@ func (m *Module) GetDownloadName() string {
 
 // GetBuildPath returns the path to the please BUILD file where this module is defined.
 func (m *Module) GetBuildPath() string {
+	currentDir, err := os.Getwd()
+	if err != nil {
+		panic(err)
+	}
+	if SingleFileBuild {
+		return fmt.Sprintf("%s/third_party/go/BUILD", currentDir)
+	}
 	splitPath := strings.Split(m.Path, "/")
 	pathMinusEnd := strings.Join(splitPath[:len(splitPath)-1], "/")
 	if splitPath[0] == "github.com" {
 		pathMinusEnd = strings.Join(splitPath[:2], "/")
 	}
-	currentDir, err := os.Getwd()
-	if err != nil {
-		panic(err)
-	}
+
 	return fmt.Sprintf("%s/third_party/go/%s/BUILD", currentDir, pathMinusEnd)
 }
 
 // GetFullyQualifiedName returns the please build target for this module.
 func (m *Module) GetFullyQualifiedName() string {
-	splitPath := strings.Split(m.Path, "/")
-	pathMinusEnd := strings.Join(splitPath[:len(splitPath)-1], "/")
-	if splitPath[0] == "github.com" {
-		pathMinusEnd = strings.Join(splitPath[:2], "/")
+	buildDir := ""
+	if !SingleFileBuild {
+		splitPath := strings.Split(m.Path, "/")
+		pathMinusEnd := strings.Join(splitPath[:len(splitPath)-1], "/")
+		if splitPath[0] == "github.com" {
+			pathMinusEnd = strings.Join(splitPath[:2], "/")
+		}
+		buildDir = fmt.Sprintf("third_party/go/%s", pathMinusEnd)
+	} else {
+		buildDir = "third_party/go"
```

Comment:
think this would be neater if we just instantiated buildDir := "third_party/go" and then modified it if not single file build?